src/test/java/edu/ucla/library/prl/harvester/services/HarvestJobSchedulerServiceIT.java

```diff
@@ -0,0 +1,263 @@
+
+            // Set up HarvestScheduleStoreService
+            final HarvestScheduleStoreService dbService = HarvestScheduleStoreService.create(dbConnectionPool);
+
+            myHarvestScheduleStoreService = binder.setAddress(HarvestScheduleStoreService.ADDRESS)
+                    .register(HarvestScheduleStoreService.class, dbService);
+            myHarvestScheduleStoreServiceProxy = HarvestScheduleStoreService.createProxy(aVertx);
+
+            // Set up HarvestService
+            final HarvestService harvestService = HarvestService.create(aVertx, config);
+
+            myHarvestService = binder.setAddress(HarvestService.ADDRESS).register(HarvestService.class, harvestService);
+
+            try {
+                myTestProviderBaseURL = new URL(config.getString(Config.TEST_PROVIDER_BASE_URL));
+            } catch (final MalformedURLException details) {
+                return Future.failedFuture(details);
+            }
+
+            return Future.succeededFuture();
+        }).onSuccess(nil -> aContext.completeNow()).onFailure(aContext::failNow);
+    }
+
+    /**
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @AfterEach
+    public void afterEach(final Vertx aVertx, final VertxTestContext aContext) {
+        final CompositeFuture wipeBackingServices =
+                CompositeFuture.all(TestUtils.wipeDatabase(myDbConnectionPool), TestUtils.wipeSolr(mySolrClient));
+
+        wipeBackingServices.compose(nil -> {
+            if (myService != null) {
+                return myService.close();
+            } else {
+                return Future.succeededFuture();
+            }
+        }).onSuccess(nil -> aContext.completeNow()).onFailure(aContext::failNow);
+    }
+
+    /**
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @AfterAll
+    public void tearDown(final Vertx aVertx, final VertxTestContext aContext) {
+        myHarvestScheduleStoreServiceProxy.close().compose(nil -> {
+            mySolrClient.shutdown();
+
+            return myDbConnectionPool.close();
+        }).onSuccess(nil -> aContext.completeNow()).onFailure(aContext::failNow);
+    }
+
+    /**
+     * Tests that a service instance triggers the near-future job(s) that were already in the database before
+     * instantiation.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @Test
+    @Timeout(value = 90, timeUnit = TimeUnit.SECONDS)
+    public void testInstantiationExistingJob(final Vertx aVertx, final VertxTestContext aContext) {
+        final Checkpoint serviceSaved = aContext.checkpoint();
+
+        // Add jobs before instantiation of the service
+        initDB().compose(numberOfJobsCreated -> {
+            final Checkpoint jobResultReceived = aContext.checkpoint(numberOfJobsCreated);
+            final EventBus eb = aVertx.eventBus();
```

Comment:
Would prefer a less cryptic name than eb (e.g. eventBus), but that's just my 2c here.